### PR TITLE
fix: escapes tilde in the cwd of the oldfiles picker

### LINKED ISSUE #2286: oldfiles picker returns empty results with tilde in cwd
### Description

My current working directory contains a `~` character as it's the iCloud folder itself on a Mac.

### Neovim version

```markdown
NVIM v0.9.0-dev-395+g3098064f3-dirty
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

macOS 12.6

### Telescope version / branch / rev

master (e960efa)

### checkhealth telescope

```markdown
==============================================================================
telescope: require("telescope.health").check()

Checking for required plugins ~
- OK plenary installed.
- OK nvim-treesitter installed.

Checking external dependencies ~
- OK rg: found ripgrep 13.0.0
- OK fd: found fd 8.6.0

===== Installed extensions ===== ~

Telescope Extension: `dap` ~
- No healthcheck provided

Telescope Extension: `frecency` ~
- OK sql.nvim installed.
- OK nvim-web-devicons installed.

Telescope Extension: `fzf` ~
- OK lib working as expected
- OK file_sorter correctly configured
- OK generic_sorter correctly configured

Telescope Extension: `ports` ~
- No healthcheck provided

Telescope Extension: `projects` ~
- No healthcheck provided
```

### Steps to reproduce

```bash
mkdir -p /tmp/test~dir
touch /tmp/test~dir/README.md
nvim --headless --noplugin -u scripts/minimal_init.vim
:e /tmp/test~dir/README.md
:cd /tmp/test~dir
:enew # the oldfiles picker hides the file if its open
:lua require("telescope.builtin").oldfiles({only_cwd=true})
```

### Expected behavior

`README.md` is visible in the list

### Actual behavior

The list is empty.

### Minimal config

```Lua
None
```

### PR DESCRIPTION
**Pull request by adamtajti**
# Description

The oldfiles picker didn't return any results in this case a tilde character was present in the path of the current working directory.

Fixes #2286

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- Manually, with an without the change
- By writing new automated tests and running `make test`

**Configuration**:
* Neovim version (nvim --version): NVIM v0.9.0-dev-395+g3098064f3-dirty
* Operating system and version: macOS 12.6

# Checklist:

- [x] My code follows the style guidelines of this project (stylua)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (lua annotations)

### CODE REVIEW

**Comment by kkharji** [comment]
Tried to reproduce the issue using the steps laid out in #2286

I got:

```
E5108: Error executing lua Vim:E874: (NFA) Could not pop the stack!
stack traceback:
        [C]: in function 'matchstrpos'
        ...tart/telescope.nvim/lua/telescope/builtin/__internal.lua:530: in function <...tart/telescope.nvim/lua/telescope/builtin/__internal.lua:529>
        vim/shared.lua: in function 'tbl_filter'
        ...tart/telescope.nvim/lua/telescope/builtin/__internal.lua:529: in function 'v'
        ...tart/telescope.nvim/lua/telescope/builtin/__internal.lua:1380: in function 'v'
        ...cker/start/telescope.nvim/lua/telescope/builtin/init.lua:516: in function 'oldfiles'
        [string ":lua"]:1: in main chunk
```

This PR indeed fix this edge case. Would merge after failing Github actions gets fixed.

**Review by adamtajti — COMMENTED** [review]
Ok, so it looks like I was missing `../plenary.nvim` and `../tree-sitter-lua` for the tests to pass. 

I faced further failures in lua parsing while I had `tree-sitter-lua` cloned locally. I checked the GitHub Workflow config and as far I understood that one doesn't setup the `tree-sitter-lua` plugin. I was able to progress after I removed it. We could possibly remove that dependency from `scripts/minimal_init.vim` if it's really not needed.

The find_files + scrolling test Job was timing out after 10s. Something seems to be up with the latest plenary job's sync call as the `on_exit` handler ran with exit code 0 after I configured it up, but the sync was still detecting a timeout. I saw that other tests are using `require "telescope.testharness"` instead of `require "telescope.pickers._test"`. After I changed to that the test passed fine. If that's alright, maybe we should remove that file as the only other reference (than itself) seems to be a commented line in `lua/tests/automated/layout_strategies_spec.lua`:1

**Comment by adamtajti** [comment]
Alright, it looks like we still have test failures. I'll get back to this in about a day and try to replicate the CI flow more closely on my machine next day as I have different outputs locally.

**Comment by adamtajti** [comment]
Fingers crossed for the next run :) I ran the CI script locally for the different versions on my MacOS 12.6, and it raised the issue with the oldfiles test (the missing emoji next to the `> ` prompt). It ran fine for the find_files tests though, so I decided to add back the skip fir these tests on the CI for now.

File: lua/telescope/pickers/_test.lua [hunk 1]
```diff
@@ -1,244 +0,0 @@
-local assert = require "luassert"
```
Comment (kkharji) [inline lua/telescope/pickers/_test.lua:1]: Is this file delete intentionally or is it debug the ci?

Comment (adamtajti) [reply lua/telescope/pickers/_test.lua:1]: I deleted this because the only usage of it that I found was [in the scrolling specs](https://github.com/nvim-telescope/telescope.nvim/pull/2287/files#diff-e28029904538b8499b83db28a26c7110e4a85cd6f618fd88ac1a11b3feca047bL3), which seems to timeout on other pull requests too. 

- [This is a recent failing run from master](https://github.com/nvim-telescope/telescope.nvim/actions/runs/3855177396/jobs/6569957658#step:6:287). 
- [These are the usages of it in the current master](https://github.com/search?q=repo%3Anvim-telescope%2Ftelescope.nvim+require+%22telescope.pickers._test%22&type=code).

As far as I see the `require("telescope.testharness").run_file` creates a process similarly like this one used to, but this one failed to get the response from the scrolling test for some reason (I haven't looked into it deeper).

Comment (kkharji) [reply lua/telescope/pickers/_test.lua:1]: cc @Conni2461 what do you think?

If it's okay with @Conni2461, this file should be deleted in a different PR called something like fix(tests) or ref(test): remove packers/_test.lua

File: lua/tests/automated/pickers/scrolling_spec.lua [hunk 2]
```diff
@@ -1,12 +0,0 @@
...
-local tester = require "telescope.pickers._test"
-
-local log = require "telescope.log"
-log.use_console = false
-
-describe("scrolling strategies", function()
-  it("should handle cycling for full list", function()
-    tester.run_file [[find_files__scrolling_descending_cycle]]
-  end)
-end)
```
Comment (Conni2461) [inline lua/tests/automated/pickers/scrolling_spec.lua:12]: uff, this test was still using the old testing framework.

can we fix it rather than deleting it?

I think it should be enough to replace the require call (we can also drop the reload und log stuff

```
local tester = require "telescope.testharness"
```

Comment (Conni2461) [reply lua/tests/automated/pickers/scrolling_spec.lua:12]: ah nvm you moved it to `find_files_spec.lua`

File: lua/tests/pickers/oldfiles__escape_tilde.lua [hunk 3]
```diff
@@ -0,0 +1,27 @@
+local helper = require "telescope.testharness.helpers"
```
Comment (Conni2461) [inline lua/tests/pickers/oldfiles__escape_tilde.lua:1]: yeah we dont need that file, if we already have a automated test. can you remove it.

Thanks :)

File: lua/telescope/pickers/_test.lua (hunk 1 above)
Comment (Conni2461) [reply lua/telescope/pickers/_test.lua:1]: this was the old testing framework and should actually be removed. Separate PR is good idea. Thanks :)

**Comment by Conni2461** [comment]
> I faced further failures in lua parsing while I had tree-sitter-lua cloned locally. I checked the GitHub Workflow config and as far I understood that one doesn't setup the tree-sitter-lua plugin. I was able to progress after I removed it. We could possibly remove that dependency from scripts/minimal_init.vim if it's really not needed.

We only need treesitter lua fro the docgen ci so we can have multiple files minimal rcs. One for testing and one for docgen. we need treesitter and plenary for docgen.

Thanks :)

**Comment by adamtajti** [comment]
https://github.com/nvim-telescope/telescope.nvim/pull/2307

File: lua/telescope/pickers/_test.lua (hunk 1 above)
Comment (adamtajti) [reply lua/telescope/pickers/_test.lua:1]: https://github.com/nvim-telescope/telescope.nvim/pull/2307
